### PR TITLE
Add marketplace extractor and query

### PR DESCRIPTION
**Pull request by nnkken**
Query: `/likechain/likenft/v1/marketplace?type={listing|offer}[&class_id=blablabla][&nft_id=blablabla][&creator=blablabla]`
Currently order by expiration, should order by price?

### CODE REVIEW

**Comment by nnkken** [comment]
Changed to order by price instead of expiration.
Turns out doesn't need to change test cases...

File: rest/marketplace_test.go [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+			name:   "listings by NFT ID",
+			query:  "type=listing&nft_id=" + nfts[1].NftId,
+			length: 1,
+			items:  []db.NftMarketplaceItem{marketplaceItems[1]},
+		},
+		{
+			name:   "offers by NFT ID",
+			query:  "type=offer&nft_id=" + nfts[1].NftId,
+			length: 1,
+			items:  []db.NftMarketplaceItem{marketplaceItems[4]},
```
Comment (williamchong) [inline rest/marketplace_test.go:207]: nft id should also come with class id for it to be an unique key

**Review by williamchong — CHANGES_REQUESTED** [review]

File: db/schema/v009.sql [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+);
+
+CREATE INDEX idx_nft_marketplace_class_id ON nft_marketplace (
+  type,
+  class_id,
+  expiration,
+  price
+);
+
+CREATE INDEX idx_nft_marketplace_nft_id ON nft_marketplace (
```
[the remark this answers is not in the document]
Comment (nnkken) [reply db/schema/v009.sql:42]: For speeding up searches with NFT ID only, not sure if we have this use case, but seems quite natural?

File: poller/poller.go [hunk 3]
```diff
@@ -136,6 +137,9 @@ func poll(pool *pgxpool.Pool, ctx *CosmosCallContext, lastHeight int64) (int64,
 			}
 		}
 	}
+	batch.UpdateLatestBlockHeight(maxHeight)
```
[the remark this answers is not in the document]
Comment (nnkken) [reply poller/poller.go:140]: No, extractor runs asynchronously with poller, and they have different "height" (`META_BLOCK_HEIGHT` for poller, `META_EXTRACTOR` for extractor)

File: db/marketplace_test.go [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+			name:   "listings by NFT ID",
+			query:  QueryNftMarketplaceItemsRequest{Type: "listing", NftId: nfts[1].NftId},
+			length: 1,
+			items:  []NftMarketplaceItem{marketplaceItems[1]},
+		},
+		{
+			name:   "offers by NFT ID",
+			query:  QueryNftMarketplaceItemsRequest{Type: "offer", NftId: nfts[1].NftId},
+			length: 1,
+			items:  []NftMarketplaceItem{marketplaceItems[4]},
```
[the remark this answers is not in the document]
Comment (nnkken) [reply db/marketplace_test.go:194]: So `NFT ID only` is not our use case?

**Review by williamchong — APPROVED** [review]

**Pull request merged**